packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManager.java

```diff
@@ -389,6 +391,47 @@ public static long measureText(
             ? paragraphAttributes.getInt(MAXIMUM_NUMBER_OF_LINES_KEY)
             : UNSET;
 
+    int numberOfLines =
+      paragraphAttributes.hasKey(NUMBER_OF_LINES_KEY)
+        ? paragraphAttributes.getInt(NUMBER_OF_LINES_KEY)
+        : UNSET;
+
+    int lines = layout.getLineCount();
+    if (numberOfLines != UNSET && numberOfLines != 0 && numberOfLines >= lines && text.length() > 0) {
+      int numberOfEmptyLines = numberOfLines - lines;
+      SpannableStringBuilder ssb = new SpannableStringBuilder();
```

Comment:
This is going to be missing all of the spans for things like font size, etc that the user will have set.

Reply:
few lines below I copy all spans